packages/steps/src/BuildWorkflow.ts

```diff
@@ -19,7 +19,18 @@ export class BuildWorkflow {
     let maybeError: Error | null = null;
     let hasAnyPreviousStepFailed = false;
     for (const step of this.buildSteps) {
-      if (step.shouldExecuteStep(hasAnyPreviousStepFailed)) {
+      let shouldExecuteStep = false;
+      try {
+        shouldExecuteStep = step.shouldExecuteStep(hasAnyPreviousStepFailed);
+      } catch (err: any) {
+        step.ctx.logger.error({ err });
+        step.ctx.logger.error(
+          `Runner failed to evaluate if it should execute step "${step.displayName}"`
+        );
```

Comment:
```suggestion
        step.ctx.logger.error(
          `Runner failed to evaluate if it should execute step "${step.displayName}", using step's if condition "${step.ifCondition}". This can be caused by trying to access non-existing object property. If you think this is a bug report it here: https://github.com/expo/eas-cli/issues.`
        );
```